Creiwry/irl-meeting-app
# Integrate conversation validation into app

### User Story
https://templatethpprojetfinal.atlassian.net/browse/THPPROJ-12?atlOrigin=eyJpIjoiZWIxOWViZTM2YWQ5NDAxZGI5YjBiMWM5OTRhZWZkYTYiLCJwIjoiaiJ9

### What does this PR do?
This PR introduces new geolocation functionality and corresponding tests to ensure the stability of the code. It updates the user's latitude and longitude based on their actions and checks the conditions to turn geolocation off. Tests have been added for both the new functionality in UsersController as well as custom methods in models and helpers. It also adds functionality to check user in on scanning area, and signing them out once they leave the geofence or they sign out. With check out, their location is also set to nil. Methods for filtering active conversations have also been added

### Where should the reviewer start?
mainly in the conversations model and controller, but also in the custom devise destroy method and update location method. a geolocation_scan_mode has been added as well. The seed has been modified so that two versions are available: one for the front end, and one for testing actual functionality. 

### How should this be manually tested?
Sign in as two different users in two browsers, scan location, and see the other appear and disappear in the users index. 

### Are there any breaking changes?
none
